hs_file_types/models/generic.py

```diff
@@ -116,6 +116,7 @@ class GenericLogicalFile(AbstractLogicalFile):
     Composite Resource """
     metadata = models.OneToOneField(GenericFileMetaData, related_name="logical_file")
     data_type = "genericData"
+    verbose_content_type = "Generic Data"  # used during discovery
```

Comment:
I think it's worth pointing out that each file_type has a static method for display name and type name introduced with #2502.

https://github.com/hydroshare/hydroshare/blob/2502-composite-resource-phase-2/hs_file_types/models/generic.py#L155  

I'm not entirely certain of your discovery requirements, but you may get some mileage out of them.

Reply:
if you need the spaces, could you use get_aggregation_type_name() and just split the camelCase?

Reply:
@sblack-usu The display name is fine, but I still need to make the non-composite resources of the same type have the same display names.

Reply:
@sblack-usu The display name is much too long for discovery, while the type name is too concise.

Reply:
@sblack-usu I would rather not rely on CamelCase, because screwing this up is going to cause dire problems with the discovery view.

Reply:
@sblack-usu Add to this that you only support a small subset of the existing resource types. I need codes for them as well, and they have to match when there's an overlap.

Reply:
@aphelionz I need to conform to the conventions in the current file type code so I need to base this PR on the composite resource PR. How close are we to merging the composite resource PR?